### ISSUE TITLE
min query length

### ISSUE THREAD

**Issue by victorwpbastos** [truckingsim/Ajax-Bootstrap-Select#43]
Is there a way to configure the min query length before trigger the search?

**Comment by truckingsim** [comment]
No but it wouldn't be too hard to implement I don't think.

**Comment by victorwpbastos** [comment]
Currently I'm working with beforeSend on ajax options. How can I update the messages in the plugin?

**Comment by truckingsim** [comment]
What do you mean by messages?

**Comment by victorwpbastos** [comment]
When I type something even if the beforeSend abort the transaction the label says: `Searching...`. I want to update this label to something like: `type at least 3 characters`

**Comment by vschettino** [comment]
Does anyone know how to do this validation based on beforeSend on ajax options? I'm trying to use ajax.beforesend returning false in case the query is too short, but it doesn't stop the request.

**Comment by flansch** [comment]
Returning *false* didn't work for me, but explicitly aborting the request:

    .ajaxSelectPicker ({
    	ajax: {
    		url: '...',
    		beforeSend: function (xhr, opts) {
    			if (this.options.data.searchterm.length < 3)
    			{
    				xhr.abort ();
    			}
    		},
    		data: function () {
    			var params = {
    				searchterm: '{{{q}}}'
    			};
    			return params;
    		}
    	}
    });

**Comment by vschettino** [comment]
@flansch's solution does work for me!
Thanks

**Comment by truckingsim** [comment]
I feel like this is the right way to do this too, so I'm going to close this and if other people search for this, use @flansch's answer.

**Comment by markcarver** [comment]
@flansch's answer is correct. However, it should be noted that aborting an XHR request will cause it to "fail" and thus trigger any error callback set. Using something like the following should help distinguish when an AJAX request is programmatically aborted: http://stackoverflow.com/a/5958734
